crates/ruff/src/rules/flake8_comprehensions/fixes.rs

```diff
@@ -552,7 +573,51 @@ pub(crate) fn fix_unnecessary_collection_call(
     };
     tree.codegen(&mut state);
 
-    Ok(Edit::range_replacement(state.to_string(), expr.range()))
+    let code = if wrap_in_spaces {
+        wrap_code_in_spaces(&state, checker, expr.range())
+    } else {
+        state.to_string()
+    };
+
+    Ok(Edit::range_replacement(code, expr.range()))
+}
+
+/// Adds spaces around the code generated from `state` if `wrap_in_spaces` is true.
+fn wrap_code_in_spaces(state: &CodegenState, checker: &Checker, expr_range: TextRange) -> String {
+    if checker.semantic_model().in_f_string() {
+        if let Some(f_string_range) = checker.indexer.f_string_range(expr_range.start()) {
+            let f_string = checker.locator.slice(f_string_range);
+
+            // find the braces that delimit the f-string
+            let f_string_start = f_string_range.start();
+            let f_string_sbrace =
+                f_string_start + TextSize::try_from(f_string.find('{').unwrap()).unwrap();
+            let f_string_ebrace =
+                f_string_start + TextSize::try_from(f_string.rfind('}').unwrap()).unwrap();
+
+            let mut buf = Vec::with_capacity(3);
```

Comment:
Nit: Can we directly push to the string instead of building a vec and then joining at the end?